### PR TITLE
Add LLD build to linux-arm64 wheel

### PR DESCRIPTION
**Pull request by tzunghanjuang**
**Context:**
Linux_arm64 for some reasons does not recognize `mlir::stablehlo::populateChloToStablehloPatterns` in ChloPasses.
Please check https://github.com/PennyLaneAI/catalyst/actions/runs/10185739889/job/28179133487?pr=977

**Reason:**
Missing LLD. https://github.com/tensorflow/mlir-hlo/issues/55

**Changes:**
The scripts for linux-arm64 do not follow the steps in Linux-x86-64 and miss lld. This PR includes LLD build back.

### CODE REVIEW

File: .github/workflows/scripts/linux_arm64/rh8/build_lld.sh [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+# Install python dependencies
+/usr/bin/python3 -m pip install pennylane pybind11 PyYAML cmake ninja
+
+cmake -S /catalyst/mlir/llvm-project/llvm -B llvm-build -G Ninja \
+      -DCMAKE_BUILD_TYPE=Release \
+      -DLLVM_TARGETS_TO_BUILD="host" \
+      -DLLVM_ENABLE_PROJECTS="lld"
+
+cmake --build /catalyst/llvm-build --target lld
```
Comment (dime10) [inline .github/workflows/scripts/linux_arm64/rh8/build_lld.sh:41]:
```suggestion
cmake --build /catalyst/llvm-build --target lld

```

File: .github/workflows/scripts/linux_arm64/rh8/build_llvm.sh [hunk 2]
```diff
@@ -43,7 +43,7 @@ cmake -S /catalyst/mlir/llvm-project/llvm -B /catalyst/llvm-build -G Ninja \
     -DLLVM_INSTALL_UTILS=ON \
     -DLLVM_ENABLE_ZLIB=OFF \
     -DLLVM_ENABLE_ZSTD=FORCE_ON \
-    -DLLVM_ENABLE_LLD=OFF \
+    -DLLVM_ENABLE_LLD=ON \
     -DMLIR_ENABLE_BINDINGS_PYTHON=ON \
     -DPython3_EXECUTABLE=/usr/bin/python3 \
     -DPython3_NumPy_INCLUDE_DIRS=/opt/_internal/cpython-${PYTHON_VERSION}.${PYTHON_SUBVERSION}/lib/python${PYTHON_VERSION}/site-packages/numpy/core/include \
```
Comment (dime10) [inline .github/workflows/scripts/linux_arm64/rh8/build_llvm.sh:49]: Should LLVM not also be built with protected symbols? (next line)

**Review by dime10 — COMMENTED** [review]
Thanks! Could you also post a successful run of the linux arm wheel action after final changes? :)

File: .github/workflows/scripts/linux_arm64/rh8/build_llvm.sh (hunk 2 above)
Comment (tzunghanjuang) [reply .github/workflows/scripts/linux_arm64/rh8/build_llvm.sh:49]: Yes. I 've added it.

**Comment by tzunghanjuang** [comment]
Linux arm64 wheel action: https://github.com/PennyLaneAI/catalyst/actions/runs/10221321390

**Review by dime10 — APPROVED** [review]
Thank you @tzunghanjuang 🥳

**Pull request merged**